### PR TITLE
UI: Add go to scene button

### PR DESCRIPTION
**Pull request by cg2121**
### Description
Add button in source context bar to go to scene of selected nested scene.

Screenshot from 2022-09-05 09-09-21

### Motivation and Context
https://ideas.obsproject.com/posts/1904/go-to-scene

### How Has This Been Tested?
Clicked button to make sure it went to the selected scene.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by CodeYan01** [comment]
Is it possible for this to have a hotkey? Like Edit Transform, I would find myself having to use this feature a lot.

**Comment by cg2121** [comment]
Updated it to be a button in the source context bar.

Screenshot from 2022-09-02 03-34-08

**Comment by CodeYan01** [comment]
Cool! Thanks

**Comment by gxalpha** [comment]
Does clicking the button immediately transition (in non-studio mode)?

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -113,6 +113,7 @@ MoveSourceDown="Move Source(s) Down"
 SourceProperties="Open Source Properties"
 SourceFilters="Open Source Filters"
 MixerToolbarMenu="Audio Mixer Menu"
+GoToScene="Go to Scene"
```
Comment (WizardCM) [inline UI/data/locale/en-US.ini:116]:
```suggestion
GoToScene="Switch to Scene"
```

**Comment by cg2121** [comment]
@gxalpha yes, it does transition to the selected scene in non studio mode.

**Comment by Fenrirthviti** [comment]
Thank you for this submission, but we don't feel this is the best way to solve this particular problem.

**Comment by CodeYan01** [comment]
> Thank you for this submission, but we don't feel this is the best way to solve this particular problem.

not even the initial attempt where they only added a context menu item rather than a button?

**Comment by RytoEX** [comment]
> > Thank you for this submission, but we don't feel this is the best way to solve this particular problem.
> 
> not even the initial attempt where they only added a context menu item rather than a button?

Items are no longer allowed to be added _only_ to the context/right-click menu.

**Comment by CodeYan01** [comment]
oh okay, thanks for the clarification

**Pull request closed without merging**